gensim/utils.py

```diff
@@ -1709,6 +1709,29 @@ def prune_vocab(vocab, min_reduce, trim_rule=None):
     return result
 
 
+def trim_vocab_by_freq(vocab, topk, trim_rule=None):
+    """Retain `topk` most frequent words in `vocab`.
+    If there are more words with the same frequency as `topk`-th one, they will be dropped.
+    Modifies `vocab` in place, returns nothing.
+
+    Parameters
+    ----------
+    vocab : dict
+        Input dictionary.
+    topk : int
+        Number of words with highest frequencies to keep.
+    trim_rule : function, optional
+        Function for trimming entities from vocab, default behaviour is `vocab[w] <= min_count`.
+
+    """
+    if topk >= len(vocab):
+        return
+
+    sorted_vocab = sorted(vocab.keys(), key=lambda word: vocab[word], reverse=True)
```

Comment:
This looks inefficient, creating two extra copy of all keys: one in `keys()` (which should be `iterkeys()`), another in `sorted_vocab`. As I wrote before, the whole point of these vocabulary trimming dances is to save memory, otherwise it loses the point.

Reply:
How do you suggest I can eliminate the second copy, `sorted_vocab` variable? The first one is easy: `vocab.keys() -> iterkeys(vocab)`, what about the second?

Reply:
`heapq.nlargest` over a generator of integer counts (not strings) is one option. That still creates a copy, but its size depends only on `topk`, not `len(vocab)`.

Reply:
@piskvorky Fixed.